From 5570e5f298bd48d55c2756fea025f0e9c89b7997 Mon Sep 17 00:00:00 2001
From: George Dunlap <george.dunlap@citrix.com>
Date: Tue, 6 Mar 2018 16:23:32 +0100
Subject: [PATCH] grant: Release domain lock on 'map' path in cache_flush

common/grant_table.c:cache_flush() grabs the rcu lock for the current
domain, but only releases it on error paths.

Note that this is not a security issue, as the preempt count is used
exclusively for assertions at the moment.

Signed-off-by: George Dunlap <george.dunlap@citrix.com>
Reviewed-by: Jan Beulich <jbeulich@suse.com>
master commit: 156b29fca10fd25065fc501eb4b47cff931086f2
master date: 2018-02-27 11:19:27 +0000
---
 xen/common/grant_table.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xen/common/grant_table.c b/xen/common/grant_table.c
index 9ddd4f1e71..8e9437d3f8 100644
--- a/xen/common/grant_table.c
+++ b/xen/common/grant_table.c
@@ -3124,6 +3124,7 @@ static int __gnttab_cache_flush(const gnttab_cache_flush_t *cflush,
 
     unmap_domain_page(v);
     put_page(page);
+    rcu_unlock_domain(d);
 
     return ret;
 }
-- 
2.14.1

